tool_quality_checks/importer.py

```diff
@@ -6,18 +6,83 @@
 import logging
 from osgeo import ogr
 from os.path import basename
+from OASDGLDatachecker.tool_quality_checks.correct_import_file import (
+    correct_vector_layer,
+)
 
 logger = logging.getLogger(__name__)
 ogr.UseExceptions()
 
 
-def copy2pg_database(settings, in_filepath, layer_name, schema="public"):
+def importer(db, settings):
+    """
+        Loads your input files into the database for checks
+    """
+    # check if relevant parameters are there:
+    if not hasattr(settings, "manhole_layer") or not hasattr(settings, "pipe_layer"):
+        logger.error("One of the input file path is missing, like manhole or pip")
+        raise AttributeError()
+
+    # initialize source schema
+    db.create_schema("src")
+
+    import_file_based_on_filetype(
+        db, settings, settings.manhole_layer, "putten_" + settings.import_type
+    )
+    import_file_based_on_filetype(
+        db, settings, settings.pipe_layer, "leidingen_" + settings.import_type
+    )
+
+    if settings.import_type == "gbi":
+        sql_relpath = os.path.join("sql", "sql_gbi_to_3di.sql")
+        sql_abspath = os.path.join(
+            os.path.abspath(os.path.dirname(__file__)), sql_relpath
+        )
+        db.execute_sql_file(sql_abspath)
+
+
+def import_file_based_on_filetype(db, settings, file_path, out_name):
+    """
+        Check file type and send request to copy to database function
+    """
+
+    if not os.path.isfile(file_path):
+        logger.error("File %s does not exists on your computer" % file_path)
+        raise FileNotFoundError()
+
+    # prepare file import
+    file_with_extention = basename(file_path)
+    filename, file_extension = os.path.splitext(file_with_extention)
+
+    if file_extension == ".shp":
+        copy2pg_database(settings, file_path, filename, out_name, schema="src")
+    else:
+        logger.error(
+            "File extension of %s is not supported by this tool, please use .shp"
+            % file_path
+        )
+        raise AttributeError()
+
+
+def copy2pg_database(settings, in_filepath, in_name, out_name, schema="public"):
+    """
+        copy a shapefile from your drive to the database
+    """
+
+    # currently only working for shapefile
+    # TODO change naming to also include sqlite and gpkg
     datasource = set_ogr_connection_pg_database(settings)
     in_source = set_ogr_connection(in_filepath)
-    in_name = os.path.splitext(basename(in_filepath))[0]
     in_layer = in_source.GetLayerByName(in_name)
     in_srid = in_layer.GetSpatialRef()
 
+    # correct vector layer to solve issues and stuff
+    # correct_in_source, correct_layer_name = in_layer, out_name
+    correct_in_source, correct_layer_name = correct_vector_layer(
```

Comment:
`correct_layer_name` is de kloppende naam van de laag.
Maar wat is `correct_in_source`? Even verderop heb je ook `correct_in_layer = correct_in_source(...)`.
Daar klinkt "correct" meer als het werkwoord "korrigeer_dit".
Even kijken of het duidelijker kan? (Of dat ik niet moet zeuren).